packages/vue-app/template/store.js

```diff
@@ -3,130 +3,136 @@ import Vuex from 'vuex'
+    const moduleData = requireModule(path, { isState: moduleName === 'state' })
+
+    // If path is a known Vuex property
+    if (VUEX_PROPERTIES.includes(moduleName)) {
+      const property = moduleName
+      const storeModule = getStoreModule(store, namespaces, { isProperty: true })
+
+      // Replace state since it's a function
+      mergeProperty(storeModule, moduleData, property)
+      continue
     }
 
-    for (const filename of filenames) {
-      let name = filename.replace(/^\.\//, '').replace(/\.(<%= extensions %>)$/, '')
-      if (name === 'index') continue
-
-      const namePath = name.split(/\//)
-
-      name = namePath[namePath.length - 1]
-      if (['state', 'getters', 'actions', 'mutations'].includes(name)) {
-        const module = getModuleNamespace(storeData, namePath, true)
-        appendModule(module, filename, name)
-        continue
-      }
-
-      // If file is foo/index.js, it should be saved as foo
-      const isIndex = (name === 'index')
-      if (isIndex) {
-        namePath.pop()
-      }
-
-      const module = getModuleNamespace(storeData, namePath)
-      const fileModule = getModule(filename)
-
-      name = namePath.pop()
-      module[name] = module[name] || {}
-
-      // if file is foo.js, existing properties take priority
-      // because it's the least specific case
-      if (!isIndex) {
-        module[name] = Object.assign({}, fileModule, module[name])
-        module[name].namespaced = true
-        continue
-      }
-
-      // if file is foo/index.js we want to overwrite properties from foo.js
-      // but not from appended mods like foo/actions.js
-      const appendedMods = {}
-      if (module[name].appends) {
-        appendedMods.appends = module[name].appends
-        for (const append of module[name].appends) {
-          appendedMods[append] = module[name][append]
-        }
-      }
-
-      module[name] = Object.assign({}, module[name], fileModule, appendedMods)
-      module[name].namespaced = true
+    // If file is foo/index.js, it should be saved as foo
+    const isIndexModule = (moduleName === 'index')
+    if (isIndexModule) {
+      namespaces.pop()
+      moduleName = namespaces[namespaces.length - 1]
     }
-    // If the environment supports hot reloading...
-    <% if (isDev) { %>
-    if (process.client && module.hot) {
-      // Whenever any Vuex module is updated...
-      module.hot.accept(files.id, () => {
-        // Update `root.modules` with the latest definitions.
-        updateModules()
-        // Trigger a hot update in the store.
-        window.<%= globals.nuxt %>.$store.hotUpdate(storeData)
-      })
-    }<% } %>
-  } else {
-    const log = (process.server ? require('consola') : console)
-    log.warn('Classic mode for store/ is deprecated and will be removed in Nuxt 3.')
+
+    const storeModule = getStoreModule(store, namespaces)
+
+    VUEX_PROPERTIES.forEach((property) => mergeProperty(storeModule, moduleData[property], property))
```

Comment:
```js
for (const property of VUEX_PROPERTIES) {
  mergeProperty(storeModule, moduleData[property], property))
}
```

Friendly reminder of our preference for breaking across multiple lines for readability :)

Reply:
From my POV this line is fine/readable enough :man_shrugging: :relaxed: